Add logger options for color and logging line numbers

nrfox opened the pull request:
### Describe the change

Quality of life improvement. Adds the ability to enable color for logging and adds line numbers when logs are trace level. If you want to see an example of how the log options are used take a look at the local or offline PRs: https://github.com/kiali/kiali/pull/8543/files#diff-ab967ab1a2f3a1b769106eeb7bfe892ef0e81d1d27811fa15be08e6749feee1fR48-R59

### Steps to test the PR

1. Deploy Kiali/PR
2. Enable trace logs
3. See line numbers
  ```
  2025-07-29T18:33:38Z INF cmd/server.go:133 > Waiting for cluster: cluster-default cache to sync
  2025-07-29T18:33:38Z INF cmd/server.go:139 > All caches synced
  2025-07-29T18:33:38Z DBG business/controlplane_monitor.go:146 > Starting polling istiod(s) every 20 seconds for proxy status
  2025-07-29T18:33:38Z TRC business/controlplane_monitor.go:63 > Scraping istiod for debug info
  ```

### Automation testing

Unit tests included.

### Issue reference

N/A

jmazzitelli reviewed (dismissed):
not sure what's up with the linter errors in code that isn't touched here

jmazzitelli commented:
linter PR approved - once that's green and merged, you can rebase here

nrfox commented:
I forgot to `git add` a new package I added. Unfortunately the linter error was very opaque but at least we updated the dep now :grimacing:. Hopefully the issue is fixed.
